src/main/java/org/jenkinsci/plugins/docker/workflow/RegistryEndpointStep.java

```diff
@@ -105,6 +111,28 @@ public DescriptorImpl() {
             return true;
         }
 
+        @Override public UninstantiatedDescribable uninstantiate(Step step) throws UnsupportedOperationException {
+            RegistryEndpointStep s = (RegistryEndpointStep) step;
+            Map<String, Object> args = new TreeMap<>();
+            args.put("url", s.registry.getUrl());
+            args.put("credentialsId", s.registry.getCredentialsId());
+            args.put("toolName", s.toolName);
+            args.values().removeAll(Collections.singleton(null));
+            return new UninstantiatedDescribable(args);
+        }
+
+        @Override public Step newInstance(Map<String, Object> arguments) throws Exception {
+            if (arguments.containsKey("url") || arguments.containsKey("credentialsId")) {
```

Comment:
I think you mean url && credentialsId?  Or perhaps you need a handling for one but not the other?

Reply:
This is what I meant. You may have a `url` and/or a `credentialsId`.